tests/unit/assessment/test_sequencing.py

```diff
@@ -0,0 +1,74 @@
+from unittest.mock import create_autospec
+
+from databricks.sdk.service import iam, jobs
+from databricks.sdk.service.compute import ClusterDetails
+
+from databricks.labs.ucx.assessment.sequencing import MigrationSequencer, MigrationStep
+from databricks.labs.ucx.framework.owners import AdministratorLocator, AdministratorFinder
+from databricks.labs.ucx.source_code.base import CurrentSessionState
+from databricks.labs.ucx.source_code.graph import DependencyGraph
+from databricks.labs.ucx.source_code.jobs import WorkflowTask
+
+
+def test_sequence_steps_from_job_task_with_cluster(ws, simple_dependency_resolver, mock_path_lookup) -> None:
+    """Sequence a job with a task referencing a cluster.
+
+    Sequence:  # TODO: @JCZuurmond: Would expect cluster first.
```

Comment:
@nfx and @ericvergnaud : What order do we expect?

If I would go about it, I would:
1. Cluster
2. Task
3. Job

Not sure though, depends a bit on the changes the cluster requires